Sources/Kafka/RDKafka/RDKafkaClient.swift

```diff
@@ -626,19 +618,15 @@ final class RDKafkaClient: Sendable {
 
     /// Atomic assignment of partitions to consume.
     /// - Parameter topicPartitionList: Pointer to a list of topics + partition pairs.
-    func assign(topicPartitionList: RDKafkaTopicPartitionList?) throws {
-        if let topicPartitionList {
-            try topicPartitionList.withListPointer { pointer in
-                try doOrThrow {
+    func assign(topicPartitionList: RDKafkaTopicPartitionList?) async throws {
+        let result = await performBlockingCall(queue: self.gcdQueue) {
```

Comment:
Maybe it's premature optimization, but seems it's better to first unwrap the `topicPartitionList`, then do `withListPointer` and only then call `performBlockingCall`.

Reply:
Yeah, I guess this one will be changed at some point as no particular design for rebalance in swift-kafka-client yet.

Reply:
Let's change it if needed at some point